core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -699,7 +711,33 @@ else if (Tables.getTableState(context.getInstance(), table) == TableState.OFFLIN
 
     }
 
-    void addMutations(MutationSet mutationsToSend) {
+    void queueMutations(final MutationSet mutationsToSend) throws InterruptedException {
+      if (null == mutationsToSend)
+        return;
+      boolean transferred = queue.tryTransfer(new Runnable() {
+        final MutationSet m = mutationsToSend;
+
+        @Override
+        public void run() {
+          if (null != m) {
+            try {
+              addMutations(m);
+            } catch (Exception e) {
+              updateUnknownErrors("Error processing mutation set", e);
+            }
+          }
+        }
+      });
+      if (!transferred) {
+        try {
+          addMutations(mutationsToSend);
```

Comment:
I like this.  Its an interesting approach of having multiple threads bin if it happens to be busy.  All batchwriter code was written w/ assumption that one thread would bin, so this change needs careful review of all code called by binning code.